charts/backingservices/charts/srs/templates/_helpers.tpl

```diff
@@ -147,20 +154,24 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
 {{- end }}
 
-{{- define "elasticsearchBasicAuthNUsername" -}}
-{{- if  .Values.srsStorage.provisionInternalESCluster  }}
+{{- define "esDeploymentUsername" -}}
+{{- if and (.Values.srsStorage.tls.enabled) (not .Values.srsStorage.provisionInternalESCluster) (not .Values.srsStorage.basicAuthentication.enabled) (not .Values.srsStorage.awsIAM)}}
+{{- .Values.srsStorage.esCredentials.username  |  b64enc }}
+{{- else if and (.Values.srsStorage.basicAuthentication.enabled) (not .Values.srsStorage.provisionInternalESCluster) (not .Values.srsStorage.tls.enabled) }}
+{{- .Values.srsStorage.esCredentials.username  |  b64enc }}
```

Comment:
Both `if` clause and `else if` is returning same value. Can we not club them? Also there is no default `else`, will the deployment not fail if user name is set to null?

Reply:
Yes, Both return the same value.
In `if` the condition we chack is different from `else`

In `if` we check `tls is enabled`. In `else` we check if `basic authentication is enabled`.
Both can't be enabled at the same time.

Both of them when enabled need a username and password.
A default value is not required.